code/__HELPERS/icons.dm

```diff
@@ -1451,13 +1451,27 @@ GLOBAL_LIST_EMPTY(transformation_animation_objects)
 
 ///Perform a shake on an atom, resets its position afterwards
 /atom/proc/Shake(pixelshiftx = 15, pixelshifty = 15, duration = 250)
-	var/initialpixelx = pixel_x
-	var/initialpixely = pixel_y
-	var/shiftx = rand(-pixelshiftx,pixelshiftx)
-	var/shifty = rand(-pixelshifty,pixelshifty)
-	animate(src, pixel_x = pixel_x + shiftx, pixel_y = pixel_y + shifty, time = 0.2, loop = duration)
-	pixel_x = initialpixelx
-	pixel_y = initialpixely
+	var/static/list/transforms
+	if(!transforms)
+		var/matrix/first_translation = matrix()
+		var/matrix/second_translation = matrix()
+		var/matrix/third_translation = matrix()
+		var/matrix/fourth_translation = matrix()
+		first_translation.Translate(-pixelshiftx, 0)
+		second_translation.Translate(0, pixelshifty)
+		third_translation.Translate(pixelshiftx, 0)
+		fourth_translation.Translate(0, -pixelshifty)
+		transforms = list(first_translation, second_translation, third_translation, fourth_translation)
```

Comment:
so after the first call the transforms are permanent and cannot be changed? 
in addition, why drop the random nature?